src/main/java/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package service.impl;
+
+import database.Storage;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import model.FruitTransaction;
+import service.ShopService;
+import strategy.OperationStrategy;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> fruitTransactions) {
+        Map<String, Integer> toStorage = fruitTransactions.stream()
+                .collect(Collectors.toMap(
+                        FruitTransaction::getFruit,
+                        this::getQuantityOfTransaction,
+                        Integer::sum
+                ));
+        checkBalance(toStorage);
+        Storage.updateDataBase(toStorage);
```

Comment:
The `Storage.updateDataBase(toStorage);` method call in the `process()` method violates the SOLID principles. The `ShopServiceImpl` class should not be responsible for updating the `Storage` class. This responsibility should be delegated to a separate service.